diff --git a/shaders/src/surface_shading_model_cloth.fs b/shaders/src/surface_shading_model_cloth.fs
index 8168491..a71cd8c 100644
--- a/shaders/src/surface_shading_model_cloth.fs
+++ b/shaders/src/surface_shading_model_cloth.fs
@@ -49,7 +49,7 @@ vec3 surfaceShading(const PixelParams pixel, const Light light, float occlusion)
     color *= light.colorIntensity.rgb * (light.colorIntensity.w * light.attenuation * occlusion);
 #else
 #if defined(HAS_COLORED_PENUMBRA)
-    vec3 penumbraColor = min(vec3(1.0 / PI), Fd / (2.0 * (1.0 - pixel.diffuseColor)));
+    vec3 penumbraColor = min(vec3(1.0 / PI), Fd / max(2.0 * (1.0 - pixel.diffuseColor), vec3(1e-4)));
     Fd = mix(penumbraColor, Fd, pow4(occlusion));
 #endif
 
diff --git a/shaders/src/surface_shading_model_standard.fs b/shaders/src/surface_shading_model_standard.fs
index 4835c9d..523076e 100644
--- a/shaders/src/surface_shading_model_standard.fs
+++ b/shaders/src/surface_shading_model_standard.fs
@@ -119,7 +119,7 @@ vec3 surfaceShading(const PixelParams pixel, const Light light, float occlusion)
 
 #if defined(HAS_COLORED_PENUMBRA)
     // Saturate the diffuse color to simulate subsurface scattering
-    vec3 penumbraColor = min(vec3(1.0 / PI), Fd / (2.0 * (1.0 - pixel.diffuseColor)));
+    vec3 penumbraColor = min(vec3(1.0 / PI), Fd / max(2.0 * (1.0 - pixel.diffuseColor), vec3(1e-4)));
     Fd = mix(penumbraColor, Fd, pow4(occlusion));
 #endif
 
